pkg/v0/cmd.go

```diff
@@ -519,22 +519,34 @@ func cherryPick(ctx context.Context, logger *logrus.Entry, c internal.Commit, co
 			"-Xsubtree=staging/"+c.Repo, c.Hash,
 		))
 		if err != nil {
+			continueCherryPick := false
 			if strings.Contains(output, "vendor/modules.txt deleted in HEAD and modified in") {
+				continueCherryPick = true
 				// we remove vendor directories for everything under staging/, but some of the upstream repos have them
 				if _, err := internal.RunCommand(logger, exec.CommandContext(ctx,
 					"git", "rm", "--cached", "-r", "--ignore-unmatch", "staging/"+c.Repo+"/vendor",
 				)); err != nil {
 					return err
 				}
+			}
+			if strings.Contains(output, "Merge conflict in staging/"+c.Repo+"/go.mod") {
+				continueCherryPick = true
 				// Due to the `go mod` commands in the staging directory below, this file may have conflicts,
-				// So resolve it as "ours", and then use the `go mod` commands to update it
+				// So resolve it as "theirs" (i.e. incoming), and then use the `go mod` commands to update it
 				// Conflicts can arise due to downstream-only code in a staging directory affecting the
 				// `go mod` command results
 				if _, err := internal.RunCommand(logger, exec.CommandContext(ctx,
-					"git", "checkout", "--ours", "--", "staging/"+c.Repo+"/go.mod",
+					"git", "checkout", "--theirs", "--", "staging/"+c.Repo+"/go.mod",
```

Comment:
To clarify, `theirs` in this case is actually the downstream version?

Reply:
No, it's the upstream version - or the version we're merging in. For example, upstream updated k8s, we want to make sure we get that updated version.

From: https://git-scm.com/book/en/v2/Git-Tools-Advanced-Merging:

Git's PoV is from the branch you're working on, so when merging your feature branch into the main/master branch, your code is referenced as "theirs", since it's coming into the branch.

>Getting the three file versions is actually pretty easy. Git stores all of these versions in the index under “stages” which each have numbers associated with them. Stage 1 is the common ancestor, stage 2 is your version and stage 3 is from the `MERGE_HEAD`, the version you’re merging in (“theirs”).

>The `git checkout` command can also take `--ours` and `--theirs` options, which can be a really fast way of just choosing either one side or the other without merging things at all.

I basically had this backwards. It probably came out alright in my tests because api and operator-registry also updated k8s and were pulled in later.

Reply:
Gotcha, so the expectation here is that we grab the upstream go.mod files and then eventually we run a go mod tidy, vendor, etc. to adhere to the downstream specific changes?

Reply:
Yes. The `go mod` is later in this function.

Reply:
Just looked at the whole file and looks like a bit further down than this logic we build a list of commands to run including all the go mod commands